compendium/DeclarativeServices/test/TestComponentLifecycle.cpp

```diff
@@ -252,4 +278,101 @@ TEST_F(tServiceComponent, testDelayedComponent_LifeCycle) //DS_TOI_52
   auto sRef2 = ctxt.GetServiceReference<test::Interface2>();
   EXPECT_FALSE(static_cast<bool>(sRef2)) << "Service must not be available after it's dependency is removed";
 }
+
+//To check if error occured during a last function call(Windows Only)
+bool isErrored(const std::string functionName)
+{
+#if defined(US_PLATFORM_WINDOWS)
+    // Retrieve the system error message for the last-error code
+    LPSTR lpDisplayBuf;
+    DWORD dw = GetLastError();
+
+    if (dw == 0)
+        return false;
+
+    FormatMessage(
+        FORMAT_MESSAGE_ALLOCATE_BUFFER |
+        FORMAT_MESSAGE_FROM_SYSTEM |
+        FORMAT_MESSAGE_IGNORE_INSERTS,
+        NULL,
+        dw,
+        MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+        reinterpret_cast<LPTSTR>(&lpDisplayBuf),
+        0, NULL);
+
+    std::cerr << "\nERROR:\n" << functionName << " failed with error " << dw << ": " << lpDisplayBuf << std::endl;
+
+    SetLastError(0);
+    LocalFree(lpDisplayBuf);
+    return true;
+#else
+    std::cerr<<"isErrored("<<functionName<<") will only work for windows";
+    return true;
+#endif
+}
```

Comment:
There is already a function which does this in the `util` project, which this executable links against. Look at `Error.cpp`. no need to re-invent the wheel.